### PR TITLE
Show 3D widget only when necessary

### PR DESCRIPTION
**Pull request by wolny**
- [x] Add checkbox for toggling the 3D widget
- [x] Enable 3D widget via flag passed to the `VolumeEditor`

See https://github.com/ilastik/ilastik/issues/1666 for more details

Depends on https://github.com/ilastik/ilastik/pull/1686

out

### CODE REVIEW

**Comment by k-dominik** [comment]
ye looking good. I guess it is a good idea to keep this PR open until you integrate those changes with Carving. I suspect there will be some slight changes, e.g. synchronize the checkbox status to the `is_3d_visible` argument of the `__init__`...

**Comment by wolny** [comment]
@k-dominik I dug up the code from around Xmas which enables 3D widget only for carving workflow (we can easily add another workflow types as well). It involves some ilastik changes to pass around the necessary context information (i.e. workflow class), but that's the easiest way I found to do it without too much refactoring. Lemme know what you think ;)

**Comment by wolny** [comment]
This part is good to go IMO. Could you have another look @k-dominik? We need to merge this first, so that the https://github.com/ilastik/ilastik/pull/1686 build could pass.

File: volumina/volumeEditor.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def lastImageViewFocus(self, axis):
         self._lastImageViewFocus = axis
         self.newImageView2DFocus.emit()
 
-    def __init__( self, layerStackModel, parent, labelsink=None, crosshair=True, syncAlongAxes=(0,1)):
+    def __init__( self, layerStackModel, parent, labelsink=None, crosshair=True, add_3d_widget=False, syncAlongAxes=(0,1)):
```
Comment (k-dominik) [inline volumina/volumeEditor.py]: Naming the parameter `add_3d_widget` is a bit unfortunate; a 3D view is added in any case, it's just not visible, if I understand the code correctly. Maybe sticking to `is_3d_visible` or even more explicit like `3d_widget_visible` would be better

Comment (wolny) [reply volumina/volumeEditor.py]: totally agree!

File: volumina/view3d/overview3d.py [hunk 2]
```diff
@@ -31,13 +31,17 @@ class Overview3D(QWidget):
     reinitialized = pyqtSignal()  # TODO: this should not be necessary: remove
     dock_status_changed = pyqtSignal(bool)
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, is_3d_visible=False, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (wolny) [reply volumina/view3d/overview3d.py]: ✅

File: volumina/volumeEditor.py (hunk 1 above)
Comment (wolny) [reply volumina/volumeEditor.py]: ✅

**Comment by k-dominik** [comment]
I've looked at the code and played around with the carving workflow a bit. Works as expected, cool!

**Comment by wolny** [comment]
Thanks for the review. Cool that we fixed the 2D bug on the way 🙌

**Pull request merged**